app/views/layouts/_menu.html.erb

```diff
@@ -1,11 +1,32 @@
 <div class="collapse navbar-collapse navbar-responsive-collapse">
   <ul class="nav navbar-nav navbar-right">
-    <% cache_unless(can_administrate?(:manage, Menu), ['main_menu', I18n.locale]) do %>
-      <% global_menus = GlobalMenus.new %>
-      <%= menu_dropdown(menus: global_menus.guild_menus, title: t('model.menus.guild')) %>
-      <%= menu_dropdown(menus: global_menus.member_menus, title: t('model.menus.members')) %>
-      <%= menu_dropdown(menus: global_menus.company_menus, title: t('model.menus.companies')) %>
-      <%= menu_dropdown(menus: global_menus.contact_menus, title: t('model.menus.contact')) %>
+    <% cache(['main_menu', I18n.locale], skip_digest: true) do %>
+      <% MainMenu.includes(:translations).index.each do |m| %>
+        <% if m.mega %>
```

Comment:
Put row 5-20 in a partial called `_mega_menu` and row 22-27 in a partial called `_menu` and then create a method for the partial path in `MainMenu` and then you can just say
```ruby
render(MainMenu.includes(:translations).index)
```